siege_utilities/political/readers.py

```
@@ -0,0 +1,236 @@
+        params.append(seat_id)
+    if incumbent_agent_id is not None:
+        clauses.append("incumbent_agent_id = %s")
+        params.append(incumbent_agent_id)
+    if current_only:
+        clauses.append("term_end IS NULL")
+    if as_of_date is not None:
+        clauses.append("(term_start <= %s AND (term_end IS NULL OR term_end > %s))")
+        params.extend([as_of_date, as_of_date])
+
+    where = ("WHERE " + " AND ".join(clauses)) if clauses else ""
+    sql = f"SELECT * FROM {_qualified('office_terms')} {where} ORDER BY term_start DESC"
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)